src/test/java/com/google/starfish/services/FavoriteNoteServiceTest.java

```diff
@@ -0,0 +1,300 @@
+  private final Date thisWeek = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_WEEK);
+  private final Date thisMonth = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_MONTH);
+  private final Date allTime = favoriteNoteService.getDateBasedOnRecency(Recency.ALL_TIME);
+
+  private final String newSchool = "cornell university";
+  private final String newCourse = "cs1300";
+  private final String firstNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 1);
+  private final String secondNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 2);
+
+  private static final boolean runTests = Constants.TEST_DB_NAME.equals("starfish_test");
+
+  /** Clear DB and insert reference data */
+  @Before
+  public void prepare() throws Exception {
+    if(!runTests) throw new Exception("Wrong Test Database Name");
+    Operation operation = getBeforeTestOperation();
+    DbSetup dbSetup = new DbSetup(new DataSourceDestination(pool), operation);
+    dbSetup.launch();
+  }
+
+  /** Test to see if we can get all notes favorited by a single user */
+  @Test
+  public void testGettingFavoriteNotesByUserId() throws SQLException {
+    Note[] favoriteNotes = favoriteNoteService.getFavoriteNotesByUserId(pool, firstNewUserId);
+    assertEquals(favoriteNotes.length, 7);
+  }
+
+  /** Test to see if we can get the number of times a particular note has been favorited */
+  @Test
+  public void testGettingNumFavoritesByNoteId() throws SQLException {
+    long numFavoritesOfNoteOne = favoriteNoteService.getNumFavoritesByNoteId(pool, 1);
+    long numFavoritesOfNoteTwo = favoriteNoteService.getNumFavoritesByNoteId(pool, 2);
+    assertTrue(numFavoritesOfNoteOne == 1);
+    assertTrue(numFavoritesOfNoteTwo == 2);
+  }
+
+  /** Test to see if we can correctly retrieve the most trending notes today */
+  @Test
+  public void testGettingTrendingNotesToday() throws SQLException, Exception {
+    Object[][] trendingNotesToday = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, Recency.TODAY, null, null);
+    // We should be getting all 7 notes irrespective of whether they are trending or not
+    if (trendingNotesToday.length != 7) throw new Exception("Retrieved wrong number of notes: " + trendingNotesToday.length);
+    int numTrendingNotes = 2;
+    Note mostTrendingNote = (Note) trendingNotesToday[0][0];
+    long numFavoritesMostTrendingNote = (long) trendingNotesToday[0][1];
+    Note leastTrendingNote = (Note) trendingNotesToday[numTrendingNotes - 1][0];
+    long numFavoritesLeastTrendingNote = (long) trendingNotesToday[numTrendingNotes - 1][1];
+    long numFavoritesFirstNoteWithNoFavorite = (long) trendingNotesToday[numTrendingNotes][1];
+    assertTrue(mostTrendingNote.getId() == 2);
+    assertTrue(numFavoritesMostTrendingNote == 2);
+    assertTrue(leastTrendingNote.getId() == 1);
+    assertTrue(numFavoritesLeastTrendingNote == 1);
+    assertTrue(numFavoritesFirstNoteWithNoFavorite == 0);
```

Comment:
similar to a comment above, I think a better test would be to actually just check the array, that the note values correspond to every note you would expect in the array.

Reply:
So basically are you saying that instead of having 5 of these separate assertTrue statements, I could basically just have one that checks if some helper method returns true where the helper method checks the note values of the passed array for the given test?